modules/pricer/src/test/java/com/opengamma/strata/pricer/bond/DiscountingBillTradePricerTest.java

```diff
@@ -142,8 +142,8 @@ public class DiscountingBillTradePricerTest {
       .build();
   
   // pricers
-  private static final DiscountingBillProductPricer PRICER_PRODUCT = DiscountingBillProductPricer.DEFAULT;
   private static final DiscountingBillTradePricer PRICER_TRADE = DiscountingBillTradePricer.DEFAULT;
+  private static final DiscountingBillProductPricer PRICER_PRODUCT = PRICER_TRADE.getProductPricer();
```

Comment:
Add a test for the new "settlementDate" method.

Reply:
Added tests